hardware/arduino/sam/libraries/HID/HID.cpp

```diff
@@ -68,7 +68,7 @@ int HID_GetDescriptor(int8_t t)
 		HIDDescriptorListNode* current = rootNode;
 		int total = 0;
```

Comment:
I'm not sure, since I'm not the author of the USB stack. I guess the reason is that `USBD_SendControl` returns `-1` in case of failure.

(well, for the sake of precision, it seems that `USBD_SendControl` in the SAM core will never returns `-1` but the equivalent function in the AVR core, `USB_SendControl` does, probably this is something that has been inherited when the USB stack has been ported from AVR to SAM).

To fix this we probably should handle the error condition, and eventually return -1 if one call to `USB(D)_SendControl` fails.